SFUsat/sfu_state.h

```diff
@@ -11,15 +11,23 @@
 
 #include "sys_common.h"
 
+// create the list of state enums, and strings so we can print out the state. https://stackoverflow.com/questions/9907160/how-to-convert-enum-names-to-string-in-c
+#define FOREACH_STATE(state) \
+        state(STATE_SAFE)   \
+        state(STATE_READY)  \
+        state(STATE_LOW_POWER)   \
+        state(NUM_STATES)  \
 
-typedef enum {
-	STATE_SAFE,
-	STATE_READY,
-	STATE_LOW_POWER,
-	NUM_STATES
-} State_t;
+#define GENERATE_ENUM(ENUM) ENUM,
+#define GENERATE_STRING(STRING) #STRING,
```

Comment:
It might work for these macros to be moved into `sfu_utils.h` https://github.com/SFUSatClub/obc-firmware/pull/24/files/ddb28e8bec97898775ae04bd8aeb9387a875f94e#diff-0bd3f618b172c737a0ac6e97b17a32f9R34

Reply:
Fixed!